### PR TITLE
Add numpydoc Pre-Commit For Codice & Glows Pt.7 Last One

### PR DESCRIPTION
**Pull request by daralynnrhode**
# Add numpydoc Pre-Commit Hook for files contained within codice and glows instrument folder.

## Overview
Adding numpydoc pre-commit. Will be doing this in parts to make it more manageable.  Making any needed changes flagged by the new pre-commit ONLY in files listed within `imap_processing/codice` and `imap_processing/glows`.

This PR will have changes made in `imap_processing/codice` and `imap_processing/glows`.
The folders and files listed above should be the only ones that numpydoc pre-commit checks will be run on.
For the most part, almost all the docstrings contained in the given files had to be updated. Most were small changes, others I had to fill in details based on what I could discern.

The main portions that need checked--and most are marked with TODO Double Check or something similar beside them-- are the Parameters and Returns sections. These are the sections I had to do the most additions in.
## Updated Files

-pyproject.toml
-.pre-commit-config.yaml

Glows files
-imap_processing/glows/utils/constants.py
-imap_processing/glows/l1b/glows_l1b_data.py
-imap_processing/glows/l1b/glows_l1b.py
-imap_processing/glows/l1a/glows_l1a_data.py
-imap_processing/glows/l1a/glows_l1a.py
-imap_processing/glows/l0/glows_l0_data.py
-imap_processing/glows/l0/decom_glows.py

Codice files
-imap_processing/codice/utils.py
-imap_processing/codice/decompress.py
-imap_processing/codice/constants.py
-imap_processing/codice/codice_l1b.py
-imap_processing/codice/codice_l1a.py
-imap_processing/codice/codice_l0.py

## Testing
-Added numpydoc pre-commits to .pre-commit-config.yaml and pyproject.toml. For now, .pre-commit-config.yaml has exclusions included in it that will be periodically removed as progressed is made in updating all files.

### CODE REVIEW

File: imap_processing/codice/constants.py [hunk 1]
```diff
@@ -1,10 +1,11 @@
-"""Contains constants variables to support CoDICE processing.
+"""
+Contains constants variables to support CoDICE processing.
 
 The ``plan_id``, ``plan_step``, and ``view_id`` mentioned in this module are
 derived from the packet data.
 
-Acronyms
---------
```
Comment (bourque) [inline imap_processing/codice/constants.py:7]: I thought I was being cool by adding an `Acronyms` section. But `numpydoc` is not a fan 😞 😅

File: imap_processing/glows/l1b/glows_l1b_data.py [hunk 2]
```diff
@@ -272,7 +273,38 @@ def __post_init__(
...
+        Generate the L1B data for direct events using the inputs from InitVar.
+
+        Parameters
+        ----------
+        direct_events : np.ndarray
+            Direct events.
+        glows_ssclk_last_pps : int
+            No idea todo check.
+        catbed_heater_active : np.double
+            No idea todo check.
```
Comment (bourque) [inline imap_processing/glows/l1b/glows_l1b_data.py]: @maxinelasp should be able to tell you what to put here?

File: imap_processing/glows/l1b/glows_l1b_data.py [hunk 3]
```diff
@@ -272,7 +273,38 @@ def __post_init__(
...
+        spin_period_source : np.double
+            Spin period source.
+        glows_time_on_pps_valid : np.double
+            Glows time.
+        time_status_valid : np.double
+            Time status.
+        housekeeping_valid : np.double
+            Housekeeping.
+        is_pps_autogenerated : np.double
+            No idea todo check.
```
Comment (bourque) [inline imap_processing/glows/l1b/glows_l1b_data.py]: @maxinelasp should be able to tell you what to put here?

**Review by bourque — APPROVED** [review]
Aside from those few TODOs, this looks good to me!

Great work on getting through the entire codebase and making all of these fixes! I really appreciate it, that is a lot of tedious work.

File: imap_processing/glows/l1b/glows_l1b_data.py [hunk 4]
```diff
@@ -272,7 +273,38 @@ def __post_init__(
...
-        """Generate the L1B data for direct events using the inputs from InitVar."""
+        """
+        Generate the L1B data for direct events using the inputs from InitVar.
+
+        Parameters
+        ----------
+        direct_events : np.ndarray
+            Direct events.
+        glows_ssclk_last_pps : int
+            No idea todo check.
```
Comment (maxinelasp) [inline imap_processing/glows/l1b/glows_l1b_data.py]:
```suggestion
            Glows subsecond clock for the last PPS.
```

File: imap_processing/glows/l1b/glows_l1b_data.py (hunk 2 above)
Comment (maxinelasp) [inline imap_processing/glows/l1b/glows_l1b_data.py]:
```suggestion
            Flag if the catbed heater is active.
```

File: imap_processing/glows/l1b/glows_l1b_data.py [hunk 5]
```diff
@@ -272,7 +273,38 @@ def __post_init__(
...
+        catbed_heater_active : np.double
+            No idea todo check.
+        spin_period_valid : np.double
+            Valid spin period.
+        spin_phase_at_next_pps_valid : np.double
+            Spin phase.
```
Comment (maxinelasp) [inline imap_processing/glows/l1b/glows_l1b_data.py]:
```suggestion
            Flag indicating if the next spin phase is valid.
```

File: imap_processing/glows/l1b/glows_l1b_data.py [hunk 6]
```diff
@@ -272,7 +273,38 @@ def __post_init__(
...
+        spin_period_source : np.double
+            Spin period source.
+        glows_time_on_pps_valid : np.double
+            Glows time.
```
Comment (maxinelasp) [inline imap_processing/glows/l1b/glows_l1b_data.py]:
```suggestion
            Flag indicating if the glows time is valid.
```

File: imap_processing/glows/l1b/glows_l1b_data.py [hunk 7]
```diff
@@ -272,7 +273,38 @@ def __post_init__(
...
+        time_status_valid : np.double
+            Time status.
```
Comment (maxinelasp) [inline imap_processing/glows/l1b/glows_l1b_data.py]:
```suggestion
            Flag indicating if time status is valid.
```

File: imap_processing/glows/l1b/glows_l1b_data.py [hunk 8]
```diff
@@ -272,7 +273,38 @@ def __post_init__(
...
+        housekeeping_valid : np.double
+            Housekeeping.
```
Comment (maxinelasp) [inline imap_processing/glows/l1b/glows_l1b_data.py]:
```suggestion
            Flag indicating if housekeeping is valid.
```

File: imap_processing/glows/l1b/glows_l1b_data.py (hunk 3 above)
Comment (maxinelasp) [inline imap_processing/glows/l1b/glows_l1b_data.py]:
```suggestion
            Flag indicating if the PPS is autogenerated.
```

File: imap_processing/glows/l1b/glows_l1b_data.py [hunk 9]
```diff
@@ -272,7 +273,38 @@ def __post_init__(
...
+        is_pps_autogenerated : np.double
+            No idea todo check.
+        hv_test_in_progress : np.double
+            Hv test.
```
Comment (maxinelasp) [inline imap_processing/glows/l1b/glows_l1b_data.py]:
```suggestion
            Flag indicating if a HV (high voltage) test is in progress.
```

File: imap_processing/glows/l1b/glows_l1b_data.py [hunk 10]
```diff
@@ -272,7 +273,38 @@ def __post_init__(
...
+        pulse_test_in_progress : np.double
+            Pulse test.
```
Comment (maxinelasp) [inline imap_processing/glows/l1b/glows_l1b_data.py]:
```suggestion
           Flag indicating if a pulse test is in progress.
```

File: imap_processing/glows/l1b/glows_l1b_data.py [hunk 11]
```diff
@@ -272,7 +273,38 @@ def __post_init__(
...
+        memory_error_detected : np.double
+            Memory error detector.
```
Comment (maxinelasp) [inline imap_processing/glows/l1b/glows_l1b_data.py]:
```suggestion
            Flag indicating if a memory error is detected.
```

Sorry! I totally missed all these docs!

**Review by maxinelasp — APPROVED** [review]
LGTM, added some suggested changes to flesh out missing GLOWS docs

**Pull request merged**